freqtrade/freqai/data_drawer.py

```diff
@@ -0,0 +1,360 @@
+        elif not self.follow_mode:
+            logger.info("Could not find existing datadrawer, starting from scratch")
+        else:
+            logger.warning(
+                f"Follower could not find pair_dictionary at {self.full_path} "
+                "sending null values back to strategy"
+            )
+
+        return exists
+
+    def load_historic_predictions_from_disk(self):
+        """
+        Locate and load a previously saved historic predictions.
+        :returns:
+        exists: bool = whether or not the drawer was located
+        """
+        exists = Path(self.full_path / str("historic_predictions.pkl")).resolve().exists()
+        if exists:
+            with open(self.full_path / str("historic_predictions.pkl"), "rb") as fp:
+                self.historic_predictions = pickle.load(fp)
+            logger.info(f"Found existing historic predictions at {self.full_path}, but beware "
+                        "that statistics may be inaccurate if the bot has been offline for "
+                        "an extended period of time.")
+        elif not self.follow_mode:
+            logger.info("Could not find existing historic_predictions, starting from scratch")
+        else:
+            logger.warning(
+                f"Follower could not find historic predictions at {self.full_path} "
+                "sending null values back to strategy"
+            )
+
+        return exists
+
+    def save_historic_predictions_to_disk(self):
+        """
+        Save data drawer full of all pair model metadata in present model folder.
+        """
+        with open(self.full_path / str("historic_predictions.pkl"), "wb") as fp:
+            pickle.dump(self.historic_predictions, fp, protocol=pickle.HIGHEST_PROTOCOL)
+
+    def save_drawer_to_disk(self):
+        """
+        Save data drawer full of all pair model metadata in present model folder.
+        """
+        with open(self.full_path / str("pair_dictionary.json"), "w") as fp:
+            json.dump(self.pair_dict, fp, default=self.np_encoder)
+
+    def save_follower_dict_to_disk(self):
+        """
+        Save follower dictionary to disk (used by strategy for persistent prediction targets)
+        """
+        follower_name = self.config.get("bot_name", "follower1")
+        with open(
+            self.full_path / str("follower_dictionary-" + follower_name + ".json"), "w"
+        ) as fp:
+            json.dump(self.follower_dict, fp, default=self.np_encoder)
+
+    def create_follower_dict(self):
+        """
+        Create or dictionary for each follower to maintain unique persistent prediction targets
+        """
+        follower_name = self.config.get("bot_name", "follower1")
+        whitelist_pairs = self.config.get("exchange", {}).get("pair_whitelist")
+
+        exists = (
+            Path(self.full_path / str("follower_dictionary-" + follower_name + ".json"))
+            .resolve()
+            .exists()
+        )
```

Comment:
I think this can be simplified (a lot, actually).
explicitly using "str" is not necessary (it's already a string - and if you use fstring for the concatenation, you're guaranteed a string output, too).
`.resolve()` should be unnecessary - and `.exists()` should be replaced with `is_dir()` or `.is_file()` (depending if you expect this to be a directory or file).

I'd also extract the "path" generation into a separate function (or generated at class init) - the same segment (`Path(self.full_path / str("follower_dictionary-" + follower_name + ".json")`) is there at least 3 times (3 times visible at current scroll level! - twice in the same function) - extracting that will reduce code duplication - and with that also the potential for problems / typos.

Reply:
Good points - I will clean this up :-)